### PR TITLE
Chore/aac rename check error

### PR DESCRIPTION
**Pull request by jferrant**
Closes https://github.com/stacks-network/stacks-core/issues/6459

### CODE REVIEW

File: stackslib/src/chainstate/stacks/tests/block_construction.rs [hunk 1]
```diff
@@ -3961,7 +3961,7 @@ fn test_is_tx_problematic() {
                     expected_txids.push(contract_call_spends_too_much_tx.txid());
 
                     // for tenure_id == 4:
-                    // make a contract that, when called, will result in a CheckError at
+                    // make a contract that, when called, will result in a StaticCheckError at
                     // runtime
```
Comment (Jiloc) [inline stackslib/src/chainstate/stacks/tests/block_construction.rs]: should this be a `CheckErrorKind`? if it can throw a `StaticCheckError` at runtime I'll be desperate!

File: stackslib/src/chainstate/stacks/events.rs [hunk 2]
```diff
@@ -52,7 +52,7 @@ pub struct StacksTransactionReceipt {
     pub execution_cost: ExecutionCost,
     pub microblock_header: Option<StacksMicroblockHeader>,
     pub tx_index: u32,
-    /// This is really a string-formatted CheckError (which can't be clone()'ed)
+    /// This is really a string-formatted StaticCheckError (which can't be clone()'ed)
```
Comment (Jiloc) [inline stackslib/src/chainstate/stacks/events.rs]: This is a `CheckErrorKind`

Comment (Jiloc) [reply stackslib/src/chainstate/stacks/events.rs]: From the code it seems like it could also contain other types of `VmExectutionError`, but maybe those are edge cases

File: stackslib/src/chainstate/stacks/tests/block_construction.rs (hunk 1 above)
Comment (jferrant) [reply stackslib/src/chainstate/stacks/tests/block_construction.rs]: Took me way longer than it should have to reason out that yes you are correct. Was having a hard time following where it was actually executing/hitting the appropriate error. But you are right. Updating!

Comment (jferrant) [reply stackslib/src/chainstate/stacks/tests/block_construction.rs]: Fixed in bdb5d9da50b90c24fa8d0eafc0ae66abc58e3e91

File: stackslib/src/chainstate/stacks/events.rs (hunk 2 above)
Comment (jferrant) [reply stackslib/src/chainstate/stacks/events.rs]: Fixed in bdb5d9da50b90c24fa8d0eafc0ae66abc58e3e91

**Review by kantai — APPROVED** [review]

**Review by fdefelici — COMMENTED** [review]
LGTM!

Conflict ongoing

**Review by fdefelici — APPROVED** [review]
